src/Microsoft.DocAsCode.Plugins/ManifestItemCollection.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Plugins
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+
+    public class ManifestItemCollection : List<ManifestItem>
+    {
+        public ManifestItemCollection() { }
+
+        public ManifestItemCollection(IEnumerable<ManifestItem> collection)
+            : base(collection) { }
+
+        public new ManifestItem this[int index]
+        {
+            get { return base[index]; }
+            set { base[index] = value; }
+        }
+
+        [Obsolete]
+        public new int Capacity
+        {
+            get { return base.Capacity; }
+            set { base.Capacity = value; }
+        }
+
+        public new int Count => base.Count;
+
+        public new void Add(ManifestItem item) => base.Add(item);
+
+        public new void AddRange(IEnumerable<ManifestItem> collection) => base.AddRange(collection);
+
+        [Obsolete]
+        public new ReadOnlyCollection<ManifestItem> AsReadOnly() => base.AsReadOnly();
+
+        [Obsolete]
+        public new int BinarySearch(ManifestItem item) => base.BinarySearch(item);
+
+        [Obsolete]
+        public new int BinarySearch(ManifestItem item, IComparer<ManifestItem> comparer) => base.BinarySearch(item, comparer);
+
+        [Obsolete]
+        public new int BinarySearch(int index, int count, ManifestItem item, IComparer<ManifestItem> comparer) => base.BinarySearch(index, count, item, comparer);
```

Comment:
why these methods are obsolete. seems strange considering the following case:
```
List<ManifestItem> = new ManifestItemCollection();
```
could it revoke the method or not?